crates/stages/src/stages/tx_lookup.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    db::Transaction, exec_or_return, DatabaseIntegrityError, ExecAction, ExecInput, ExecOutput,
+    Stage, StageError, StageId, UnwindInput, UnwindOutput,
+};
+use reth_db::{
+    cursor::{DbCursorRO, DbCursorRW},
+    database::Database,
+    tables,
+    transaction::{DbTx, DbTxMut},
+};
+use tracing::*;
+
+const TRANSACTION_LOOKUP: StageId = StageId("TransactionLookup");
+
+/// The transaction lookup stage.
+///
+/// This stage walks over the bodies table, and sets the transaction hash of each transaction in a block
+/// to the correesponding `TransitionId` at each block. This is written to the [`tables::TxHashNumber`]
+/// This is used for looking up changesets via the transaction hash.
+#[derive(Debug)]
+pub struct TransactionLookupStage {
+    /// The number of table entries to commit at once
+    pub commit_threshold: u64,
+}
+
+#[async_trait::async_trait]
+impl<DB: Database> Stage<DB> for TransactionLookupStage {
+    /// Return the id of the stage
+    fn id(&self) -> StageId {
+        TRANSACTION_LOOKUP
+    }
+
+    /// Write total difficulty entries
+    async fn execute(
+        &mut self,
+        tx: &mut Transaction<'_, DB>,
+        input: ExecInput,
+    ) -> Result<ExecOutput, StageError> {
+        let ((start_block, end_block), capped) =
+            exec_or_return!(input, self.commit_threshold, "sync::stages::transaction_lookup");
+
+        debug!(target: "sync::stages::transaction_lookup", start_block, end_block, "Commencing sync");
+
+        let mut cursor_bodies = tx.cursor_read::<tables::BlockBodies>()?;
+        let mut tx_cursor = tx.cursor_write::<tables::Transactions>()?;
+        let (mut current_tx_id, _) = tx.get_next_block_ids(start_block)?;
```

Comment:
this is incorrect (it gets tx id for `start_block + 1`) and redundant since you have ids from `body.tx_id_range()`